components/Vote/Election.js

```diff
@@ -507,40 +61,300 @@ const query = gql`
           portrait
           disclosures
+          gender
+          birthday
+          biographyContent
           credentials {
             isListed
             description
           }
         }
+        postalCodeGeo {
+          countryName
+          countryCode
+          postalCode
+          lat
+          lon
+        }
       }
     }
   }
 `
 
-export default compose(
+const messageDateFormat = timeFormat('%e. %B %Y')
+
+const styles = {
+  wrapper: css({
+    width: '100%',
+    position: 'relative',
+    minHeight: 200
+  }),
+  header: css({
+    padding: '13px 0'
+  }),
+  actions: css({
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'center',
+    justifyContent: 'center'
+  }),
+  message: css({
+    padding: 20,
+    marginBottom: 20,
+    textAlign: 'center'
+  })
+}
+
+const sortNames = (c1, c2) => {
+  let name1 = c1.user.name.toUpperCase()
+  let name2 = c2.user.name.toUpperCase()
+  if (name1 < name2) {
+    return -1
+  }
+  if (name1 > name2) {
+    return 1
+  }
+  return 0
+}
+
+export const ElectionHeader = ({ children }) => {
+  return (
+    <div {...styles.header}>
+      <P>
+        <strong>{children}</strong>
+      </P>
+    </div>
+  )
+}
+
+export const ElectionActions = ({ children }) => {
+  return (
+    <Card style={{ position: 'sticky', bottom: 0, zIndex: 9 }}>
+      <div {...styles.actions}>{children}</div>
+    </Card>
+  )
+}
+
+const ElectionAlert = ({ children }) => {
+  const [colorScheme] = useColorContext()
+  return (
+    <div
+      {...styles.wrapper}
+      {...colorScheme.set('backgroundColor', 'alert')}
+      style={{ marginBottom: 30, padding: 20 }}
+    >
+      {children}
+    </div>
+  )
+}
+
+const updateVote = (election, vote) =>
+  [...election.candidacies]
+    .sort((c1, c2) => {
+      if (
+        (c1.recommendation && !c2.recommendation) ||
+        (c1.isIncumbent && !c2.isIncumbent)
+      ) {
+        return -1
+      } else if (
+        (c2.recommendation && !c1.recommendation) ||
+        (c2.isIncumbent && !c1.isIncumbent)
+      ) {
+        return 1
+      } else return sortNames(c1, c2)
+    })
+    .map(candidate => ({
+      candidate,
+      selected:
+        vote.find(v => v.candidate.id === candidate.id)?.selected || false
+    }))
+
+const Election = compose(
   voteT,
   withMe,
-  graphql(submitElectionBallotMutation, {
-    props: ({ mutate }) => ({
-      submitElectionBallot: (electionId, candidacyIds) => {
-        return mutate({
-          variables: {
-            electionId,
-            candidacyIds
-          }
-        })
-      }
-    })
-  }),
+  withAddressData
+)(
+  ({
+    election,
+    vt,
+    addressData,
+    me,
+    discussionTag,
+    mandatoryCandidates,
+    showMeta
+  }) => {
+    const electionId = election.id
+    const useGenElection = useMemo(
+      () => createPersistedState(`republik-general-election-${electionId}`),
+      [electionId]
+    )
+    const [vote, setVote] = useGenElection([])
```

Comment:
Sorry I missed that the data is mirrored into the the local storage during the review. The setup needs to be more sophisticated: only ids of candidates with a selected true or false should be persistent. The list of available candidates should always come from the server and then only be enriched with a selection from local storage.

Reply:
yeah i kinda had the same thought overnight :)

Reply:
also on privacy grounds

Reply:
jup, we should also ensure that the selection is deleted after the ballot is submitted